server/broadcast.go

```diff
@@ -722,6 +722,10 @@ func selectOrchestrator(ctx context.Context, n *core.LivepeerNode, params *core.
 			bcastOS = bcastOS.OS().NewSession(pfx)
 		}
 
+		oinfo := n.OrchestratorPool.GetInfo(tinfo.Transcoder)
```

Comment:
is it the reason why the bug was occurring only when establishing a session for the transcoding?
Because when a broadcaster first submits an orchestrator's domain, it is parsed in a way that is supposed to match the format of the `serviceAddr` - except apparently for the localhost/127.0.0.1 edge-case. In fact in the case of https://github.com/livepeer/go-livepeer/issues/2485 not prefixing the O domain `solarfarm.multiorch.com` with `https` is not a problem right away because the [`parseOrchAddrs` function](https://github.com/livepeer/go-livepeer/blob/master/cmd/livepeer/starter/starter.go#L1191-L1209) takes care of reformatting it - if so, rather than creating a new function (`ResolveURL`) can't we reuse the same util used in `starter.go` (with the appropriate changes to handle the ipv6/localhost edge-cases)?